tests/accessor/accessor_implicit_conversions.hpp

```diff
@@ -0,0 +1,381 @@
+      const src_accessor_t src_acc(data_buf, cgh);
+      return src_acc;
+    };
+    // Implicit conversion should be verified within command
+    const invoke_implicit_conversion_t modify_acc_functor;
+
+    if constexpr (Dimension == 0) {
+      check_zero_dim_constructor<AccType, SrcDataT, DstAccessMode, Target>(
+          get_acc_functor, modify_acc_functor);
+    } else {
+      const auto r = util::get_cts_object::range<Dimension>::get(1, 1, 1);
+      check_common_constructor<AccType, SrcDataT, Dimension, DstAccessMode,
+                               Target>(r, get_acc_functor, modify_acc_functor);
+    }
+  }
+
+ public:
+  void operator()(const std::string& type_name,
+                  const std::string& target_name) const {
+    const auto make_section_name = [&](sycl::access_mode SrcAccessMode,
+                                       sycl::access_mode DstAccessMode,
+                                       const std::string& details) {
+      return detail::section_name_prototype<AccType, Dimension>::get(type_name)
+          .with("target", Target)
+          .with("source access", SrcAccessMode)
+          .with("destination access", DstAccessMode)
+          .with("case", details)
+          .create();
+    };
+
+    // From read-only accessor to read-only accessor
+    {
+      constexpr auto AccessMode = sycl::access_mode::read;
+
+      SECTION((
+          make_section_name(AccessMode, AccessMode, "from 'T' to 'const T'"))) {
+        using SrcDataT = DataT;
+        using DstDataT = const DataT;
+        run_check<SrcDataT, AccessMode, DstDataT, AccessMode>();
+      }
+      SECTION((
+          make_section_name(AccessMode, AccessMode, "from 'const T' to 'T'"))) {
+        using SrcDataT = const DataT;
+        using DstDataT = DataT;
+        run_check<SrcDataT, AccessMode, DstDataT, AccessMode>();
+      }
+    }
+    // From read-write accessor to read-only accessor
+    {
+      constexpr auto SrcAccessMode = sycl::access_mode::read_write;
+      constexpr auto DstAccessMode = sycl::access_mode::read;
+
+      SECTION((make_section_name(SrcAccessMode, DstAccessMode,
+                                 "from 'T' to 'const T'"))) {
+        using SrcDataT = DataT;
+        using DstDataT = const DataT;
+        run_check<SrcDataT, SrcAccessMode, DstDataT, DstAccessMode>();
+      }
+      SECTION((make_section_name(SrcAccessMode, DstAccessMode,
+                                 "from 'const T' to 'T'"))) {
+        using SrcDataT = const DataT;
+        using DstDataT = DataT;
+        run_check<SrcDataT, SrcAccessMode, DstDataT, DstAccessMode>();
+      }
+    }
+  }
+};
+
+/**
+ * @brief Validates implicit conversion for local accessor
+ */
+template <typename DataT, typename DimensionT>
+class check_conversion_local {
+  static constexpr auto AccType = accessor_type::local_accessor;
+  static constexpr int Dimension = DimensionT::value;
+
+  static_assert(!std::is_const_v<DataT>, "No need to pass const type here");
```

Comment:
Do we need `DataT` here? `DataT` don't used enywhere into `check_conversion_local`

Reply:
No need or forbidden?

Reply:
> No need or forbidden?

I am was wrong, this is needed

Reply:
@keryell That's semantic limitation for interface, as all const-related pairs are checked within.